sql/_hny_join.sql

```diff
@@ -0,0 +1,196 @@
+/*
+DESCRIPTION:
+    Merging devdb with hny. This requires the following procedure.
+    Following _hny.sql query with HNY_matches were completed
+
+    5) Assign flags to indicate one_hny_to_many_dev and/or one_dev_to_many_hny.
+        a) handle one devdb to many hny records cases by concatenating those hny_ids and combines units/fields 
+        to create a single record
+        b) Rest of the cases (one hny to many devdb, many hny to many devdb) are handled by including every 
+        possible joins with devdb records for each hny recor.
+
+    6) Combine the two tables to create HNY_devdb
+
+INPUTS: 
+    HNY_matches (
+        hny_id,
+        job_number,
+        match_priority,
+        job_type,
+        resid_flag,
+        all_counted_units,
+		total_units
+    )
+
+    hpd_hny_units_by_building (
+        ogc_fid,
+        project_id,
+        project_name,
+        number,
+        street,
+        reporting_construction_type,
+        all_counted_units,
+        total_units,
+        one_dev_to_many_hny integer,
+        one_hny_to_many_dev integer,
+        project_start_date text,
+        project_completion_date text,
+        extremely_low_income_units numeric,
+        very_low_income_units numeric,
+        low_income_units numeric,
+        moderate_income_units numeric,
+        middle_income_units numeric,
+        other_income_units numeric,
+        studio_units numeric,
+        "1_br_units" numeric,
+        "2_br_units" numeric,
+        "3_br_units" numeric,
+        "4_br_units" numeric,
+        "5_br_units" numeric,
+        "6_br+_units" numeric,
+        unknown_br_units numeric,
+        counted_rental_units numeric,
+        counted_homeownership_units numeric
+    )
+
+    hny_geocode_results (
+        * uid,
+        geo_bin,
+        geo_bbl,
+        geo_latitude,
+        geo_longitude
+
+    )
+
+    
+OUTPUTS: 
+    HNY_devdb(
+        hny_id text,
+        job_number text,
+        classa_hnyaff text,
+        all_hny_units text,
+        one_dev_to_many_hny integer,
+        one_hny_to_many_dev integer,
+        project_start_date text,
+        project_completion_date text,
+        extremely_low_income_units numeric,
+        very_low_income_units numeric,
+        low_income_units numeric,
+        moderate_income_units numeric,
+        middle_income_units numeric,
+        other_income_units numeric,
+        studio_units numeric,
+        "1_br_units" numeric,
+        "2_br_units" numeric,
+        "3_br_units" numeric,
+        "4_br_units" numeric,
+        "5_br_units" numeric,
+        "6_br+_units" numeric,
+        unknown_br_units numeric,
+        counted_rental_units numeric,
+        counted_homeownership_units numeric
+    )
+*/
+
+-- 5) Identify relationships between devdb records and hny records
+DROP TABLE IF EXISTS HNY_devdb;
+WITH 
+	-- Find cases of many-hny-to-one-devdb, after having filtered to highest priority
```

Comment:
What does filter to highest priority mean here? The only filter I see is count >1 to find many to one relationship

Reply:
the priority has to do with in hny_matches step in `_hny_matches.sql` when the different types of matches (through bbl/geometry etc) were resolved to a single one.

Reply:
Revisiting this now - so the priority is based on hierarchy described [here](https://github.com/NYCPlanning/db-developments/blob/553-HNY-Join/sql/_hny_match.sql#L14-L24) - if so, I think adding a quick note or additional detail that this is where the priority is outlined would be useful. I think this is some of the more complex work that we do and its easiest for us when we have to revisit to have as much detailed information here